unity/Assets/Scripts/StochasticRemoteFPSAgentController.cs

```diff
@@ -65,37 +65,47 @@ public class StochasticRemoteFPSAgentController : BaseFPSAgentController {
 #endif
         }
 
-        public override void MoveRelative(ServerAction action) {
-            if (!allowHorizontalMovement && Math.Abs(action.x) > 0) {
+        public void MoveRelative(
+            float? moveMagnitude = null,
+            float x = 0f,
+            float z = 0f,
+            float noise = 0f,
+            bool forceAction = false
+        ) {
+
+            if (!moveMagnitude.HasValue) {
+                moveMagnitude = gridSize;
+            } else if (moveMagnitude.Value <= 0f) {
+                throw new InvalidOperationException("moveMagnitude must be null or >= 0.");
+            }
+
+            if (!allowHorizontalMovement && Math.Abs(x) > 0) {
                 throw new InvalidOperationException("Controller does not support horizontal movement. Set AllowHorizontalMovement to true on the Controller.");
             }
-            var moveLocal = new Vector3(action.x, 0, action.z);
-            var moveMagnitude = moveLocal.magnitude;
-            if (moveMagnitude > 0.00001) {
-                // random.NextGaussian(RotateGaussianMu, RotateGaussianSigma);
-                var random = new System.Random();
+
+            var moveLocal = new Vector3(x, 0, z);
+            float xzMag = moveLocal.magnitude;
+            if (xzMag > 0.00001) {
```

Comment:
Perhaps `0.00001` should have an `f` at the end for float?

Reply:
Technically not my code but fair enough, I changed it to 1e-5.

Reply:
That too, but `1e-5` still makes it a double instead of a float. I added a small change in https://github.com/allenai/ai2thor/pull/803/commits/f78edae25668549627726f3e32cc6d5f116e070a, which simply makes `1e-5` into `1e-5f`.